handlers/routeservice.go

```diff
@@ -135,6 +144,89 @@ func (r *RouteService) ServeHTTP(rw http.ResponseWriter, req *http.Request, next
 	next(rw, req)
 }
 
+// createAllowlistPatterns collects the static parts of wildcard allowlist expressions and wildcards stripped of their first segment.
+//
+// Each entry is checked to follow DNS wildcard notation, e.g.
+// *.domain-123.com, subdomain.example.com
+//
+// but not subdomain.*.example.com, *.*.example.com or invalid DNS names, e.g. ?!.example.com
+//
+// This function is exported so it can be tested as part of the route service test
+func CreateAllowlistPatterns(allowlist []string) (map[string]struct{}, error) {
+
+	// This check is a preliminary configuration check and case insensitive. Route URL host names are matched verbatim.
+	var validAllowlistEntryPattern = regexp.MustCompile(`(?i)^(\*\.)?[a-z\d-]+(\.[a-z\d-]+)+$`)
+
+	// allowlistHostNames is a hash set containing DNS names or DNS name suffixes based on wildcards.
+	// map[string]struct{} is used for minimal memory overhead for the value, using the map as set.
+	var allowlistHostNames = make(map[string]struct{}, len(allowlist))
+
+	for _, entry := range allowlist {
+
+		if !validAllowlistEntryPattern.MatchString(entry) {
+			return nil, fmt.Errorf("invalid route service hairpinning allowlist entry: %s. Must be wildcard (*.domain.com) or FQDN (subdomain.domain.com)", entry)
+		}
+
+		hostName := entry
+
+		if strings.HasPrefix(entry, "*") {
+			// strip wildcard, leave the rest of the FQDN, including leading '.'
+			hostName = stripHostFromFQDN(entry)
+		}
+		allowlistHostNames[hostName] = struct{}{}
+	}
+
+	return allowlistHostNames, nil
+}
+
+// stripHostFromFQDN strips the host, i.e. first segment, from a fully qualified domain name
+func stripHostFromFQDN(entry string) string {
+	splitString := strings.SplitN(entry, ".", 2)
+	return "." + splitString[1]
+}
```

Comment:
```suggestion
func sanitizeFQDN(entry string) string {
                 sanitizedEntry := entry
		if strings.HasPrefix(entry, "*") {
			// strip wildcard, leave the rest of the FQDN, including leading '.'
			splitString := strings.SplitN(entry, ".", 2)
			sanitizedEntry = "." + splitString[1]
		}
	
	return sanitizedEntry
}
```

Reply:
also make sure to lower-case the entry before adding to the map, so that
"myhost.mydomain.com"
and
"MyHost.MyDOMAIN.com"
become the same map key.